### PR TITLE
[5730] RSYNC test tool

### PR DESCRIPTION
**Pull request by konangc87**
# Shared modules utils Generic Message Dispatcher class.

## **Related Issue**
[5730](https://github.com/wazuh/wazuh/issues/5730)

## Description
This issue aims to create a test tool to stress rsync and check that it's working correctly.

## DoD / Tasks
- [X] Implementation

### CODE REVIEW

File: src/shared_modules/rsync/testtool/agentEmulator.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * August 28, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include <iostream>
+// #include <chrono>
```
Comment (tinachoB) [inline src/shared_modules/rsync/testtool/agentEmulator.cpp:12]: remove unnecessary code

File: src/shared_modules/rsync/testtool/agentEmulator.cpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+#include <iomanip>
+#include <ctime>
+#include <cstdlib>
+#include "agentEmulator.h"
+
+
+
+constexpr auto CRETE_STATEMENT{ "CREATE TABLE data_values(`key` BIGINT, `data1` BIGINT, `data2` BIGINT, `data3` BIGINT, PRIMARY KEY (`key`)) WITHOUT ROWID;"};
```
Comment (tinachoB) [inline src/shared_modules/rsync/testtool/agentEmulator.cpp:20]: Typo -> CREATE_STATEMENT

File: src/shared_modules/rsync/testtool/agentEmulator.cpp [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+static void callback(const ReturnTypeCallback /*type*/,
+                     const cJSON* /*json*/,
+                     void* /*ctx*/)
+{
+    // const std::unique_ptr<char, CJsonDeleter> spJsonBytes{ cJSON_PrintUnformatted(json) };
+    // std::cout << spJsonBytes.get() << std::endl;
+}
+
+AgentEmulator::AgentEmulator(const std::chrono::milliseconds updatePeriod,
+							 const unsigned int maxDbItems,
```
Comment (tinachoB) [inline src/shared_modules/rsync/testtool/agentEmulator.cpp:60]: indentation issues

File: src/shared_modules/rsync/testtool/agentEmulator.cpp [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+		{
+			"Error creating rsync handle."
+		};
+	}
+	sync_callback_data_t callback{agentEmulatorSyncCallback, this};
+	if (rsync_register_sync_id(m_rsyncHandle, m_agentId.c_str(), m_dbSyncHandle, m_config, callback))
+	{
+		throw std::runtime_error
+		{
+			"Error creating rsync handle."
```
Comment (tinachoB) [inline src/shared_modules/rsync/testtool/agentEmulator.cpp:84]: Error registering ID

File: src/shared_modules/rsync/testtool/cmdArgsHelper.h [hunk 5]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Wazuh DBSYNC
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 02, 2020.
```
Comment (tinachoB) [inline src/shared_modules/rsync/testtool/cmdArgsHelper.h:4]: Wrong date

File: src/shared_modules/rsync/testtool/main.cpp [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        };
+        const std::chrono::milliseconds syncPeriod{100};
+        const auto maxDbItems{100};
+        AgentEmulator agent{syncPeriod, maxDbItems, syncQueue, args.outputFolder()};
+        ManagerEmulator manager{syncQueue};
+        while(getc(stdin) != 'q');
+    }
+    catch(const std::exception& ex)
+    {
+        std::cout << ex.what() << std::endl;
```
Comment (tinachoB) [inline src/shared_modules/rsync/testtool/main.cpp:44]: Please, add `CmdLineArgs::showHelp();` call here.

**Review by tinachoB — CHANGES_REQUESTED** [review]

**Review by tinachoB — APPROVED** [review]
:+1:

**Review by Dwordcito — CHANGES_REQUESTED** [review]

File: src/shared_modules/rsync/testtool/agentEmulator.cpp [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    {
+        throw std::runtime_error
+        {
+            "Error creating dbsync handle."
+        };
+    }
+    return handle;
+}
+
+struct smartDeleterJson
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply src/shared_modules/rsync/testtool/agentEmulator.cpp:34]: And please, use Upper case for struct/class names. smartDeleterJson -> SmartDeleterJson

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**